converter/internal/otelcolconvert/testdata/span.yaml

```diff
@@ -0,0 +1,33 @@
+receivers:
+  otlp:
+    protocols:
+      grpc:
+      http:
+
+exporters:
+  otlp:
+    # Our defaults have drifted from upstream, so we explicitly set our
+    # defaults below (balancer_name and queue_size).
+    endpoint: database:4317
+    balancer_name: pick_first
+    sending_queue:
+      queue_size: 5000
+
+processors:
+  span:
```

Comment:
Given the amount of type casting being down here, and it's not obvious to me how much of the code paths are being hit by default, maybe we should explicitly define all the options here so we have more code coverage? 

(Probably would need a comment justifying why this test case has so many options when the others just use the default settings)

Reply:
Yeah, I tested a complete configuration manually just to verify I didn't get any panics.

Will update the testdata file, add a relevant comment and clean up the conversions.